src/riscv-mode-integration.adoc

```diff
@@ -29,9 +29,8 @@ include::img/cap-encoding-xlen32-mode.edn[]
 [#cap_encoding_xlen64_mode]
 include::img/cap-encoding-xlen64-mode.edn[]
 
-{cheri_mode_ext_name} allows the M bit to be set to 1 when the capability does
-not grant <<x_perm>>. In this case, the M bit is superfluous, so the encoding
-may be used to support additional features in future extensions.
+For RV64, {cheri_mode_ext_name} allows the M bit to be set to 1 when the capability does not grant <<x_perm>>.
+In this case, the M bit is superfluous, so the encoding may be used to support additional features in future extensions.
```

Comment:
If we unified permissions between RV32 and RV64 these peculiarities could also go away.

Reply:
this is true - but I worry that we will be unnecessarily limiting the options for RV64 where there is ample encoding space, so we shoudl allow more flexibility.
We just need to define the quirkly cases.

Reply:
https://github.com/riscv/riscv-cheri/issues/164

Reply:
I think if there were such motivating use cases, they should be equally applicable to RV32 and we could hopefully argue for their existence now.

I can of course see us missing something. But there is also nothing stopping us from adding in extensions to add permission combinations back to RV64 (or in RV32, if not too many) if we discover they were needed. It is worse to take them back.

But I still think the default position should be to save the bits, and to align base specs for software trans-portability.

Reply:
We discussed that briefly in yesterdays SIG/TG meeting and I don't think we need to constrain RV64 (yet). For RV32 we have no choice, but for RV64 we might as well allow these permission combinations since adding them back later is a lot more difficult than defining a new restricted encoding once we add new features that want to make use of those bits.

Reply:
I agree with @LawrenceEsswood here. We now have a different behavior for acperm depending on XLENMAX, from an instruction that was meant to abstract the permissions handling from the permissions encoding. We could be in the position where we will have to write different permissions code for 32 and 64 bit CHERI (and accidentally using that code written for 64 bit on a 32 bit system could result in unexpected behavior), or else write trans-portable code that will restrict the permission combinations to those of the 32 bit degenerative case anyway.

Do we really want to have diverging base CHERI permission behavior between 32 bit and 64 bit at this stage?

Reply:
Do we want it? No. But it's inevitable given the constraints. RV64 should not be restricted in its combinations, but RV32 has to be due to encoding space limitations.